packages/odata2ts/src/generator/ModelGenerator.ts

```diff
@@ -46,7 +58,8 @@ class ModelGenerator {
   }
 
   private generateModels() {
-    this.dataModel.getModels().forEach((model) => {
+    const modelTypes = reorderModelsByInheritance<ModelType>(this.dataModel.getModels());
```

Comment:
Instead of doing the ordering when generating models or q-objects, I could imagine that this would be better placed within the `DataModel` class, e.g. via the method `getModels` or a special method like `getModelsOrdered` or something like that.

You could also use the `postProcess` method to do this ordering only once and store this order in the `DataModel`.

Reply:
That was my first attempt but at that point I didn't want to touch the interface generation. Agree that's a better place to but it.